src/Compiler/CMakeLists.txt

```diff
@@ -66,9 +66,21 @@ add_onnx_mlir_library(CompilerPasses
   )
 
 if (ONNX_MLIR_VENDOR)
-  set_source_files_properties(CompilerUtils.cpp
-    PROPERTIES COMPILE_DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR} \"")
+  list(APPEND DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR}\"")
 endif()
+if (PRODUCT_VERSION)
```

Comment:
I'd like to see a test for this, after this change there's logic that is not being validated anywhere

Reply:
This relies on if we decide to have default values or not.

Reply:
User settable variables should all start with ONNX_MLIR so I'd use ONNX_MLIR_PRODUCT_VERSION or simply ONNX_MLIR_VERSION instead.

Reply:
Renamed to `ONNX_MLIR_PRODUCT_VERSION`.

Reply:
> User settable variables should all start with ONNX_MLIR 

That is very good practice, thanks for enforcing it.